projects/project_router.js

```diff
@@ -0,0 +1,83 @@
+const express = require("express");
+
+const Projects = require("./project_model");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+  try {
+    const projects = await Projects.find();
+    res.json(projects);
+  } catch (err) {
+    res.status(500).json({ message: "Failed to get projects" });
+  }
+});
+
+router.get("/:id/actions", async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const actions = await Projects.findProjectAction(id);
+
+    if (actions.length) {
+      res.json(actions);
+    } else {
+      res
+        .status(404)
+        .json({ message: "Could not find actions for given project" });
+    }
+  } catch (err) {
+    res.status(500).json({ message: "Failed to get actions" });
+  }
+});
+
+router.post("/", async (req, res) => {
+  const projectData = req.body;
+
+  try {
+    const project = await Projects.addProject(projectData);
+    res.status(201).json(project);
+  } catch (err) {
+    res.status(500).json({ message: "Failed to create new project" });
+  }
+});
+
+router.post("/:id/actions", async (req, res) => {
```

Comment:
Another beautiful endpoint and sorry you couldn't see it work. The problem is not in the endpoint though, it's in your data access function `.addAction()`.... check my comments in the data access file.